### ISSUE TITLE
Creating  `azurerm_kubernetes_cluster` is enabling windows profile by default though is not mentioned in config

### ISSUE THREAD

**Issue by adeshreddy** [terraform-providers/terraform-provider-azurerm#6358]
Community Note
Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
If you are interested in working on this issue or have submitted a pull request, please leave a comment

Terraform (and AzureRM Provider) Version
$ terraform version
+ Terraform v0.12.24
+ provider.azurerm v2.1.0

Affected Resource(s)
`azurerm_kubernetes_cluster`
`azurerm_kubernetes_cluster_node_pool`

Terraform Configuration Files

```
resource "azurerm_resource_group" "aks" {
  location = "EAST US 2"
  name     = "FRIENDLY-rg"
  tags     = local.common_tags
}

resource "azurerm_kubernetes_cluster" "aks" {
  name                = "friendlyaks"
  location            = azurerm_resource_group.aks.location
  resource_group_name = azurerm_resource_group.aks.name
  dns_prefix          = "aks"
  kubernetes_version  = "1.15.7"
  node_resource_group = "aks-rg"
  default_node_pool {
    name                = "aks"
    vm_size             = "Standard_D2_v2"
    type                = "VirtualMachineScaleSets"
    enable_auto_scaling = "true"
    os_disk_size_gb     = 30
    vnet_subnet_id      = data.azurerm_subnet.subnet.id
    max_count           = "6"
    min_count           = "3"

  }
  tags = local.common_tags
  service_principal {
    client_id     = var.client_id
    client_secret = var.client_secret
  }
  role_based_access_control {
    enabled = true
  }
  network_profile {
    network_plugin    = "azure"
    load_balancer_sku = "Standard"
  }
}

resource "azurerm_kubernetes_cluster_node_pool" "aks_nodepool" {
  name                  = "nodepool1"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.aks.id
  vm_size               = "Standard_DS2_v2"
  max_count             = 6
  min_count             = 3
  enable_auto_scaling   = "true"
  node_labels = {
    app = "test"
    env = "dev"
  }
  node_taints    = ["key=value:NoSchedule"]
  vnet_subnet_id = data.azurerm_subnet.subnet.id
}

//This should be added after an initial apply 
resource "azurerm_kubernetes_cluster_node_pool" "aks_nodepool_2" {
  name                  = "nodepool2"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.aks.id
  vm_size               = "Standard_DS2_v2"
  max_count             = 6
  min_count             = 3
  enable_auto_scaling   = "true"
  node_labels = {
    app = "test"
    env = "dev"
  }
  node_taints    = ["key=value:NoSchedule"]
  vnet_subnet_id = data.azurerm_subnet.subnet.id
}
```

Expected Behavior
Currently, I am provisioning the AKS cluster with the existing configuration. I am adding multiple node pools. Whenever I add new node pool resource it should just add a set of min VM's to the existing cluster.
`resource "azurerm_kubernetes_cluster_node_pool" "aks_nodepool_2"` is added or removed from configuration a  node pool should be added or destroyed accordingly to the existing cluster. Which can be done through the  Azure portal 

Actual Behavior
When `resource "azurerm_kubernetes_cluster_node_pool" "aks_nodepool_2"` is added to the configuration and do the terraform plan. Plan say's it will force destroying the cluster and recreate it. 

Steps to Reproduce
+ terraform plan
+ terraform apply
 This will create a whole new cluster 
now add `resource "azurerm_kubernetes_cluster_node_pool" "aks_nodepool_2"`
+ terraform plan
https://github.com/terraform-providers/terraform-provider-azurerm/issues/3971

**Comment by adeshreddy** [comment]
I had dug more into the issue and found out though we are not assigning windows profile in AKS args.
it's creating windows profile with below-mentioned args and saving it in state file which is the reason it is trying to force destroy and create resources again since aks.tf and state file are not in sync.
`{
  "admin_password": "",
  "admin_username": "azureuser"
}`

**Comment by adeshreddy** [comment]
#6235

**Comment by asubmani** [comment]
I ran into this issue as well. I am not explicitly defining all default actions e.g. `type: "VirtualMachineScalesets"`   as it seems defaults are constantly changing.

**Comment by tombuildsstuff** [comment]
👋🏼

This perpetual diff is unfortunately due to a breaking behavioural change in the AKS API where all new clusters now get a Windows Profile configured even if one isn’t specified. There’s a fix available for this in v2.5 of the Azure Provider or later (we’re currently on v2.6 at the time of writing).

Since this should be fixed by updating to the latest version of the Azure Provider I’m going to close this issue for the moment, but please let us know if you’re still having issues after updating and we’ll take another look.

Thanks!
